app/scripts/models/mixins/resume-token.js

```diff
@@ -44,16 +54,47 @@ define(function (require, exports, module) {
     },
 
     /**
-     * Sets model properties from a ResumeToken.
+     * Sets model properties from a ResumeToken, unless said token
+     * is invalid according to `this.resumeTokenSchema`.
      *
      * @method populateFromResumeToken
      * @param {ResumeToken} resumeToken
      */
     populateFromResumeToken: function (resumeToken) {
       if (this.resumeTokenFields) {
-        this.set(resumeToken.pick(this.resumeTokenFields));
+        var pickedResumeToken = resumeToken.pick(this.resumeTokenFields);
+
+        if (isResumeTokenValid.call(this, pickedResumeToken)) {
+          this.set(pickedResumeToken);
+        }
       }
     }
   };
+
+  function isResumeTokenValid (resumeToken) {
+    if (this.resumeTokenSchema) {
+      var error = vat.validate(resumeToken, this.resumeTokenSchema).error;
+
+      if (error) {
+        if (this.sentryMetrics) {
+          if (error instanceof ReferenceError) {
+            error = authErrors.toMissingResumeTokenPropertyError(error.key);
+          } else {
+            error = authErrors.toInvalidResumeTokenPropertyError(error.key);
+          }
+
+          // HACK: Interpolate the invalid property name into the error
+          //       message. One day this will be handled automagically.
+          error.message = authErrors.toInterpolatedMessage(error);
```

Comment:
Copying over [this comment]:

Can we set the message and capture to sentry outside of this function, perhaps in populateFromResumeToken? Otherwsie we are assuming isResumeTokenValid is called at most once per resume token and could end up with multiple reports if that assumption is ever broken.